stats/stats/src/charts/lines/new_accounts.rs

```diff
@@ -1,114 +1,157 @@
-                        b.timestamp != to_timestamp(0) AND 
-                        b.consensus = true
-                    ORDER BY t.from_address_hash, b.timestamp
-                ) first_tx
-                GROUP BY first_tx.date;
-                "#,
-            vec![],
-        );
-
-        let mut data = DateValueInt::find_by_statement(stmnt)
-            .all(blockscout)
+        let query = NewAccountsStatement::get_statement(range.clone());
+        let mut data = DateValueInt::find_by_statement(query)
+            .all(cx.blockscout)
             .await
             .map_err(UpdateError::BlockscoutDB)?;
-        data.sort_by_key(|v| v.date);
+        // make sure that it's sorted
+        data.sort_by_key(|d| d.date);
+        if let Some(range) = range {
+            let range = range.start().date_naive()..=range.end().date_naive();
+            trim_out_of_range_sorted(&mut data, range);
+        }
         Ok(data)
     }
 }
 
-#[async_trait]
-impl ChartFullUpdater for NewAccounts {
-    async fn get_values(
-        &self,
-        blockscout: &DatabaseConnection,
-    ) -> Result<Vec<DateValue>, UpdateError> {
-        let mut cache = self.cache.lock().await;
-        Ok(cache
-            .get_or_update(async move { Self::read_values(blockscout).await })
-            .await?
-            .into_iter()
-            .map(DateValue::from)
-            .collect())
-    }
+/// Note:  The intended strategy is to update whole range at once, even
+/// though the implementation allows batching. The batching was done
+/// to simplify interface of the data source.
+///
+/// Thus, use max batch size in the dependant data sources.
+pub type NewAccountsRemote = RemoteDatabaseSource<NewAccountsQueryBehaviour>;
+
+pub type NewAccountsRemoteString = MapToString<NewAccountsRemote>;
+
+pub struct Properties;
+
+impl Named for Properties {
+    const NAME: &'static str = "newAccounts";
 }
 
-#[async_trait]
-impl crate::Chart for NewAccounts {
-    fn name(&self) -> &str {
-        "newAccounts"
-    }
-    fn chart_type(&self) -> ChartType {
+impl ChartProperties for Properties {
+    fn chart_type() -> ChartType {
         ChartType::Line
     }
 }
 
-#[async_trait]
-impl ChartUpdater for NewAccounts {
-    async fn update_values(
-        &self,
-        db: &DatabaseConnection,
-        blockscout: &DatabaseConnection,
-        current_time: chrono::DateTime<chrono::Utc>,
-        force_full: bool,
-    ) -> Result<(), UpdateError> {
-        self.update_with_values(db, blockscout, current_time, force_full)
-            .await
-    }
-}
+pub type NewAccounts = LocalDbChartSource<
+    NewAccountsRemoteString,
+    (),
+    DefaultCreate<Properties>,
+    BatchUpdate<
+        NewAccountsRemoteString,
+        (),
+        PassVecStep,
+        // see `NewAccountsRemote` docs
+        BatchMax,
+        Properties,
+    >,
+    DefaultQueryVec<Properties>,
+    Properties,
+>;
+
+pub type NewAccountsInt = MapParseTo<NewAccounts, DateValueInt>;
```

Comment:
It looks strange that you query Int, then cast it to String and then cast it to String again, is it intended?

Reply:
Apparently, it is queried as int and not string. I copied the logic from original implementation. Changing it to string (in query) results in:
```
thread 'charts::lines::new_accounts::tests::update_new_accounts' panicked at stats/src/tests/simple_test.rs:41:38:
called `Result::unwrap()` on an `Err` value: BlockscoutDB(Query(SqlxError(ColumnDecode { index: "\"value\"", source: "mismatched types; Rust type `core::option::Option<alloc::string::String>` (as SQL type `TEXT`) is not compatible with SQL type `INT8`" })))
```